src/atama/sender/node/sender_node.cpp

```diff
@@ -33,13 +33,92 @@ using namespace std::chrono_literals;
 namespace atama::sender
 {
 
-SenderNode::SenderNode(rclcpp::Node::SharedPtr node, std::shared_ptr<atama::head::Head> head)
-: node(node), head(head)
+SenderNode::SenderNode(
+  rclcpp::Node::SharedPtr node,
+  std::shared_ptr<atama::head::Head> head, bool & done_get_joints)
+: node(node), head(head), is_done_get_joints_data(&done_get_joints)
 {
   set_joints_publisher = node->create_publisher<tachimawari_interfaces::msg::SetJoints>(
     get_node_prefix() + "/set_joints", 10);
   set_head_publisher = node->create_publisher<atama_interfaces::msg::Head>(
     get_node_prefix() + "/set_head_data", 10);
+
+  {
+    using atama_interfaces::srv::RunHead;
+    run_head_service = node->create_service<RunHead>(
+      get_node_prefix() + "/run_head",
+      [this](std::shared_ptr<RunHead::Request> request,
+      std::shared_ptr<RunHead::Response> response) {
+        rclcpp::Rate rcl_rate(8ms);
+
+        // Assume the function is not exist
+        bool is_function_exist = false;
+
+        // Check existence of the function
+        switch (request->function_id) {
```

Comment:
Since there's a possibility that `request` is a `nullptr`, let's do a check first. And please check other lines too.

Reply:
ok I will put if guard